presto-orc/src/main/java/com/facebook/presto/orc/reader/StreamReader.java

```diff
@@ -24,6 +25,12 @@
     void readBatch(Object vector)
             throws IOException;
 
+    default void readBatch(Type type, Object vector)
```

Comment:
I have confirmed that this is doable. But doing that would mean the caller of `type.createBlockBuilder` has completely no information on how many items to expect in the `BlockBuilder`.